### PR TITLE
move pluralization extraction to a separate function

### PR DESCRIPTION
**Pull request by AEnterprise**
This allows people using the this to provide their own function that handles pluralization rules for languages currently not supported by this implementation

this is optional and the current implementation is used as default

### CODE REVIEW

**Comment by pl12133** [comment]
Fixes https://github.com/synacor/preact-i18n/issues/2

@AEnterprise thanks for working on this!

File: src/lib/translate.js [hunk 1]
```diff
@@ -2,42 +2,51 @@ import delve from 'dlv';
...
+ *    @param {Number} [plural]        Indicates a quantity, used to trigger pluralization
+ *    @param {String|Array} [fallback]    Text to return if no translation is found
+ *    @param provider {function: String | false} Provider function to extract the plural form from the dictionary
+ *    @returns {String} translated
+ */
+export default function translate(id, scope, dictionary, fields, plural, fallback, provider = defaultProvider) {
+	if (scope) id = scope + '.' + id;
+
+	let value = dictionary && delve(dictionary, id);
+	value = provider(value, plural) || defaultProvider(value, plural);
```
Comment (pl12133) [inline src/lib/translate.js]: Since we are using the name `Provider` in `IntlProvider` this name might be confusing, maybe we can call it a "pluralizer"?

**Comment by AEnterprise** [comment]
oops seems i messed up a few things in the merge, i'll just squash and force push it

**Comment by AEnterprise** [comment]
i gave up on making the unit tests pass though, these are pretty much impossible to run locally on windows or WSL here are some of the things i ran into with it:

- linter failing on code i didn't touch, claiming a var that is used on the next line isn't used
- karma failing to start cause a missing plugins block in the karma config
- karma complaining about not finding a module i loaded
- karma starting to instantly die with a "1 load error" and no more info
- all modules used in the package.json commands need to be installed globally for them to be found

any help with making these tests actually run would be much appreciated but all of this involves tools that i never heard of before that just spit out non descriptive or just wrong errors

**Comment by AEnterprise** [comment]
reworking this with a new PR

**Pull request closed without merging**